src/preview/upload.rs

```diff
@@ -188,15 +183,15 @@ fn unauthenticated_upload(target: &Target) -> Result<Preview, failure::Error> {
     // KV namespaces are not supported by the preview service unless you authenticate
     // so we omit them and provide the user with a little guidance. We don't error out, though,
     // because there are valid workarounds for this for testing purposes.
-    let script_upload_form = if target.kv_namespaces.is_some() {
+    let script_upload_form = if !target.kv_namespaces.is_empty() {
         message::warn(
             "KV Namespaces are not supported in preview without setting API credentials and account_id",
         );
         let mut target = target.clone();
-        target.kv_namespaces = None;
-        upload::form::build_preview(&target, None)?
+        target.kv_namespaces = Vec::new();
+        upload::form::build(&target, None)?
     } else {
-        upload::form::build_preview(&target, None)?
+        upload::form::build(&target, None)?
```

Comment:
i wonder if we should just do the warning check in its own if and then assign `script_upload_form` outside of the conditional. doesn't look like we need the conditional here since both arms return the same fn call